src/history/hs_rec.c

```diff
@@ -911,25 +923,69 @@ __hs_delete_reinsert_from_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, ui
          * the cell. The cell's start timestamp can be cleared during reconciliation if it is
          * globally visible.
          */
-        if (hs_start_ts >= ts || twp->stop_ts >= ts)
+        if (hs_start_ts >= ts || twp->stop_ts >= ts) {
+            if (hs_start_ts == 0 && twp->stop_ts == 0 && ts == 0)
+                *non_ts_updates = true;
             break;
+        }
     }
-    if (ret == WT_NOTFOUND)
-        return (0);
-    WT_ERR(ret);
+    WT_RET(ret);
 
     /*
      * If we find a key with a timestamp larger than or equal to the specified timestamp then the
      * specified timestamp must be mixed mode.
      */
     WT_ASSERT_ALWAYS(
       session, ts == 1 || ts == WT_TS_NONE, "out-of-order timestamp update detected");
+    return (ret);
+}
+
+/*
+ * __hs_delete_reinsert_from_pos --
+ *     Delete updates in the history store if the start timestamp of the update is larger or equal
```

Comment:
We should note down that we expect the history cursor to have been positioned when calling this function.

Reply:
Updated the comment here [2b180e8](https://github.com/wiredtiger/wiredtiger/pull/10686/commits/2b180e83db0c5ee2f52a7de9ce66b5a63924bdf0)